packages/lib/modules/transactions/transaction-steps/receipts/receipt-parsers.ts

```diff
@@ -149,6 +152,39 @@ export function parseSwapReceipt({
   }
 }
 
+export function parseLstStakeReceipt({ receiptLogs, userAddress, chain, getToken }: ParseProps) {
+  const receivedToken: HumanTokenAmount[] = getIncomingLogs(receiptLogs, userAddress).map(log => {
+    const tokenDecimals = getToken(log.address, chain)?.decimals
+    return _toHumanAmount(log.address, log.args.value, tokenDecimals)
+  })
+
+  return {
+    receivedToken,
+  }
+}
+
+export function parseLstUnstakeReceipt({ receiptLogs, userAddress, chain, getToken }: ParseProps) {
+  const receivedToken: HumanTokenAmount[] = getIncomingLogs(receiptLogs, userAddress).map(log => {
+    const tokenDecimals = getToken(log.address, chain)?.decimals
+    return _toHumanAmount(log.address, log.args.value, tokenDecimals)
+  })
+
+  return {
+    receivedToken,
+  }
+}
+
+export function parseLstWithdrawReceipt({ receiptLogs, userAddress, chain, getToken }: ParseProps) {
+  const receivedToken: HumanTokenAmount[] = getIncomingLogs(receiptLogs, userAddress).map(log => {
+    const tokenDecimals = getToken(log.address, chain)?.decimals
+    return _toHumanAmount(log.address, log.args.value, tokenDecimals)
+  })
+
+  return {
+    receivedToken,
+  }
+}
```

Comment:
You shouldn't need to parse receipts for staking/unstaking because there is no
slippage. The only reason we do receipt states in the add/remove/swap flow is
because the amounts can change between preview and confirmation. I think it's
likely all this can be removed.

Reply:
for the unstaking you are right, there is no token transferred at that stage so i have removed it

for staking and withdrawing though, tokens are transferred and i have updated the parsers to correctly parse the logs for that
i also updated the tests for there 2 parsers